docs/gyakorlat/ef/index.md

```diff
@@ -176,6 +176,84 @@ Debugger segítségével nézd meg, hogy milyen SQL utasítás generálódik: az
     }
     ```
 
+??? example "Megoldás method chain-nel"
+    ```csharp
+    using (var ctx = new AdatvezEntities())
+    {
+
+        //2.1
+        Console.WriteLine("\t2.1:");
+        var products = ctx.Product.Where(p => p.Stock > 30)
+            .Select(p => new { p.Name, p.Stock})
+            .ToList();
+
+        foreach (var item in products)
+        {
+            Console.WriteLine($"name: {item.Name}, stock: {item.Stock}");
+        };
+
+
+        //2.2
+        Console.WriteLine("\t2.2:");
+        var products2 = ctx.Product.Where(p => p.OrderItem.Count > 1)
+                                   .ToList();
+
+        foreach (var item in products2)
+        {
+            Console.WriteLine($"name: {item.Name}, orderNum: {item.OrderItem.Count}");
+        };
+
+
+        //2.3
+        Console.WriteLine("\t2.3:");
+        var products3 = ctx.Order
+            .Include(o => o.OrderItem)
+                .Include(o => o.OrderItem.Select(oi => oi.Product))
+                .Include(o => o.CustomerSite.MainCustomer)
+            .Where(o => o.OrderItem.Sum(oi => (oi.Amount * oi.Price)) > 30000)
+            .ToList();
+
+        foreach (var item in products3)
+        {
+            Console.WriteLine($"name: {item.CustomerSite.MainCustomer.Name}");
+
+            foreach (var oi in item.OrderItem)
+            {
+                Console.WriteLine($"\tname: {oi.Product.Name}, {oi.Amount}*{oi.Price}={oi.Amount*oi.Price}");
+
+            }
+        };
+
+
+        //2.4
+        Console.WriteLine("\t2.4:");
+        var maxPriceProd = ctx.Product
+            .Where(p => p.Price == ctx.Product.Max(p2 => p2.Price))
+                                              .FirstOrDefault();
+
+        Console.WriteLine($"{maxPriceProd.Name}");
+
+
+        //2.5
+        Console.WriteLine("\t2.5:");
```

Comment:
Ebben a megoldásban nem látok érdemi különbséget a ToList kivételével, ami nem szükséges. Ezt is hagyjuk ki.